paraconf/include/paraconf/PC_node.h

```diff
@@ -0,0 +1,267 @@
+/*******************************************************************************
+ * Copyright (C) 2021 Institute of Bioorganic Chemistry Polish Academy of Science (PSNC)
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * * Redistributions of source code must retain the above copyright
+ *   notice, this list of conditions and the following disclaimer.
+ * * Redistributions in binary form must reproduce the above copyright
+ *   notice, this list of conditions and the following disclaimer in the
+ *   documentation and/or other materials provided with the distribution.
+ * * Neither the name of CEA nor the names of its contributors may be used to
+ *   endorse or promote products derived from this software without specific
+ *   prior written permission.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ ******************************************************************************/
+
+#ifndef PC_NODE_H_
+#define PC_NODE_H_
+
+#include <exception>
+#include <yaml-cpp/yaml.h>
+
+#include "paraconf.h"
+#include "paraconf/error.h"
+#include "paraconf_export.h"
+
+/// Paraconf node containing yaml-cpp node
+struct PARACONF_EXPORT PC_node {
+private:
+	/// Node from yaml-cpp library
+	YAML::Node m_node;
+
+	/// Filename if yaml was loaded from file
+	std::string m_filename;
+
+	/// Acquired referenced to this node
+	std::list<PC_node> m_acquired;
+
+	/** Checks if returned node is tagged as include
+	 *  and sets the same filename
+	 * 
+	 * \param[in] node node to postprocess
+	 * 
+	 * \return postprocessed node
+	 */
+	PC_node postprocess_result(PC_node node) const;
+
+public:
+	/// PC_node iterator
+	class Iterator {
```

Comment:
Why not just wrap a `YAML::Node::iterator`?

Reply:
Because then you are losing information about filename of the node (it is only stored in paraconf)